src/use_case/save_preferences/SavePreferencesDataAccessInterface.java

```diff
@@ -0,0 +1,5 @@
+package use_case.save_preferences;
+
+public interface SavePreferencesDataAccessInterface {
+    void savePreferences();
```

Comment:
Change the function signature to `void saveUserPreference(String username, UserPreference preference)`